crates/bdk/src/wallet/coin_selection.rs

```diff
@@ -276,7 +276,7 @@ impl CoinSelectionAlgorithm for OldestFirstCoinSelection {
 pub fn decide_change(remaining_amount: u64, fee_rate: FeeRate, drain_script: &Script) -> Excess {
     // drain_output_len = size(len(script_pubkey)) + len(script_pubkey) + size(output_value)
     let drain_output_len = serialize(drain_script).len() + 8usize;
-    let change_fee = fee_rate.fee_vb(drain_output_len);
+    let change_fee = (fee_rate * Weight::from_wu((drain_output_len * 4) as u64)).to_sat();
```

Comment:
Why are we hardcoding 4 here?
Might be worth a comment...

Reply:
```rust
let change_fee = (fee_rate * Weight::from_vb_unchecked(drain_output_len as u64)).to_sat();
```